aws/scenarios/vm/agentinstall/install.go

```diff
@@ -0,0 +1,74 @@
+package agentinstall
+
+import (
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/aws"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/os"
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func Install(runner *command.Runner, env aws.Environment, params *Params, os os.OS) error {
+	agentCommand := &agentCommand{version: params.version}
+	os.Visit(agentCommand)
+	cmd := agentCommand.cmd
+	lastCommand, err := runner.Command(
+		env.CommonNamer.ResourceName("agent-install", utils.StrHash(cmd)),
+		&command.CommandArgs{
+			Create: pulumi.Sprintf(cmd, params.apiKey),
+		})
+	if err != nil {
+		return err
+	}
+
+	agentConfig := ""
+	if params.agentConfig != "" {
+		fileManager := command.NewFileManager(runner)
+		agentConfig = fmt.Sprintf(params.agentConfig, params.apiKey)
+		remotePath := os.GetConfigPath()
+		lastCommand, err = fileManager.CopyInlineFile("agent-config", agentConfig, remotePath, true, pulumi.DependsOn([]pulumi.Resource{lastCommand}))
+		if err != nil {
+			return err
+		}
+
+	}
+
+	// When the file content has changed, make sure the Agent is restarted.
+	serviceManager := os.GetServiceManager()
+	startAgentRes := env.CommonNamer.ResourceName("start-agent", utils.StrHash(serviceManager.StartAgentCmd(), agentConfig))
+	_, err = runner.Command(
+		startAgentRes,
+		&command.CommandArgs{
+			Create: pulumi.String(serviceManager.StartAgentCmd()),
+		}, pulumi.DependsOn([]pulumi.Resource{lastCommand}))
+
+	return err
+}
+
+type agentCommand struct {
+	version version
+	cmd     string
+}
+
+func (a *agentCommand) VisitUnix()    { a.cmd = getInstallFormalString("install_script.sh", a.version) }
+func (a *agentCommand) VisitMacOS()   { a.cmd = getInstallFormalString("install_mac_os.sh", a.version) }
+func (a *agentCommand) VisitWindows() { panic("Not implemented") }
+
+func getInstallFormalString(scriptName string, version version) string {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Cool !